spark/src/main/scala/geotrellis/spark/filter/ToSpatial.scala

```diff
@@ -90,4 +93,22 @@ object ToSpatial {
         )
     }
   }
+
+  def apply[
+    K: ClassTag: SpatialComponent: TemporalComponent: λ[α => M[α] => Functor[M, α]]: λ[α => Component[M[α], Bounds[α]]],
+    V: ClassTag,
+    M[_]
+  ](rdd: RDD[(K, V)] with Metadata[M[K]], ensureUnique: Boolean): RDD[(SpatialKey, V)] with Metadata[M[SpatialKey]] = {
+    val metadata = rdd.metadata.map(_.getComponent[SpatialKey])
+    val rdd2 = ensureUnique match {
+      case true =>
+        rdd
+          .map({ case (k, v) => (k.getComponent[SpatialKey], v) })
+          .map(identity).reduceByKey({ case (left, right) => left })
```

Comment:
This is only one of the possible arguments to `reduceByKey`, other more common cases would be to reduce by key using `Mean` or `Max` tile functions. To minimize API surprise I would suggest to use the method signatures to disambiguate the cases where merge is not required:


```scala
abstract class SpaceTimeToSpatialMethods[...](...) {
  def toSpatial(): RDD[(SpatialKey, V)] with Metadata[M[SpatialKey]]
  def toSpatial(instant: Long): RDD[(SpatialKey, V)] with Metadata[M[SpatialKey]]
  def toSpatial(dateTime: ZonedDateTime): RDD[(SpatialKey, V)] with Metadata[M[SpatialKey]]
}

abstract class SpaceTimeToSpatialMergeMethods[...](...) {
  def toSpatial(merge: (V, V) => V): RDD[(SpatialKey, V)] with Metadata[M[SpatialKey]]
  def toSpatial(instant: Long, merge: (V, V) => V): RDD[(SpatialKey, V)] with Metadata[M[SpatialKey]]
  def toSpatial(dateTime: ZonedDateTime, merge: (V, V) => V): RDD[(SpatialKey, V)] with Metadata[M[SpatialKey]]
}
```

Also if we're going to delegate calls to `reduceByKey` to be responsible we should expose at least the overload with partitioner parameter: https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/rdd/PairRDDFunctions.scala#L306

Reply:
I was not able to fully understand everything written above, but I have added the capability to specify a merge function and/or a partitioner.  The methods have to have a name different from `toSpatial` because if they are given that name, they hide the methods given in the other class.  A second class had to be created to avoid changing the API.